src/context.js

```diff
@@ -63,20 +64,36 @@ type ConsumerProps = {
 class LoadsConsumer extends React.Component<ConsumerProps> {
   render = () => {
     const { cacheKey, cacheProvider: localCacheProvider, children } = this.props;
-
     return (
       <Consumer>
-        {context => {
-          const cacheProvider = localCacheProvider || context.globalCacheProvider;
-          let cachedData = context.data[cacheKey];
-          if (cacheProvider && cacheProvider.get) {
-            cachedData = cacheProvider.get(cacheKey);
-          }
-          return children({
-            cache: cachedData,
-            setResponse: data => context.setResponse({ cacheKey, cacheProvider, data })
-          });
-        }}
+        {context => (
+          <Component
+            initialState={{
+              cachedData: context.data[cacheKey],
+              cacheProvider: localCacheProvider || context.globalCacheProvider,
+              hasLoaded: false
+            }}
+            didMount={({ state: { cacheProvider }, setState }) => {
+              if (cacheProvider && cacheProvider.get) {
+                const cacheResponse = cacheProvider.get(cacheKey);
+                if (cacheResponse instanceof Promise) {
+                  return cacheResponse.then(cachedData => setState({ cachedData, hasLoaded: true }));
```

Comment:
If the provided promise is rejected then setState never gets called here, leading hasLoaded to stay false and nothing gets rendered (deadlocked?).

Reply:
sad times